lightning/src/ln/functional_tests.rs

```diff
@@ -9133,6 +9133,71 @@ fn test_invalid_funding_tx() {
 	mine_transaction(&nodes[1], &spend_tx);
 }
 
+#[test]
+fn test_coinbase_funding_tx() {
+	// Miners are able to fund channels directly from coinbase transactions, however
+	// by consensus rules, outputs of a coinbase transaction are encumbered by a 100
+	// block maturity timelock. To ensure that a (non-0conf) channel like this is enforceable
+	// on-chain, the minimum depth is updated to 100 blocks for coinbase funding transactions.
+	//
+	// Note that 0conf channels with coinbase funding transactions are unaffected and are
+	// immediately operational after opening.
```

Comment:
Would it be worth adding the coinbase case to the 0conf test? FWIW it doesn't look like too much work since we already have `functional_test_utils::open_zero_conf_channel`

Reply:
Yeah, we thought it would be good to add them, so will be doing that definitely in this PR. Maybe some closes on those channels before maturity too.